eodhp_utils/pulsar/client.py

```diff
@@ -0,0 +1,52 @@
+import json
+import os
+
+from pulsar import Client as PulsarClient
+
+
+class Client:
+    def __init__(self, url=None):
+        if url is None:
+            url = os.environ.get("PULSAR_URL")
+        self.client = PulsarClient(url)
+
+
+class Producer(Client):
+    def __init__(self, topic, producer):
+        super().__init__()
+
+        self.producer = self.client.create_producer(topic=topic, producer_name=producer)
+
+    def send(self, message: bytes):
+        """Send a message. preferred format is bytes but also converts str and dict"""
+        if isinstance(message, dict):
+            message = (json.dumps(message)).encode("utf-8")
+        elif isinstance(message, str):
+            message = message.encode("utf-8")
+
+        return self.producer.send(message)
+
+    def close(self):
+        """Close client and producer"""
+        self.producer.close()
+        self.client.close()
```

Comment:
if you're going down the subclass route then I would have a close method on Client that does self.client.close() and call it from here with super().close(). This keeps all related functionality in the same level of inheritance and means if that behaviour gets any more complex you only have to update it in one place.